### ISSUE TITLE
"Limit bereits erreicht" does not disappear & no risk checks

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#3943]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Galaxy S21
- Android version: 11
- App version: 2.7.1  

## Describe the bug

A user tried if he could trick CovPass by setting the date into the future and seeing a full vaccination status. He saw that it works & set his time back to the correct date. 
Now he opened the Corona-Warn-App and was welcomed with the message "Limit bereits erreicht" and the message that the last check was at the 28.08.2021, the date he selected for his test. 
Now, this is actually two issues:
1. The "Limit bereits erreicht" message does not disappear anymore since the date of his test (06.08.2021) and
2. No new risk checks -> #3629

## Steps to reproduce the issue

1. Set the time of the phone to a date in the future
2. Open the CWA and let it carry out a risk check or let the CWA carry out a check in the background
3. Set the time back to the correct time
4. Experience the above described behaviour

## Expected behaviour

The "Limit bereits erreicht" message should disappear after 24h and risk checks should be carried out.

## Additional information
This bug report is based on the information a Twitter user shared with me in https://twitter.com/ralphspricht/status/1428022920463388674?s=20. You can also find some screenshots there.

----
Internal Tracking-ID: [EXPOSUREAPP-9077](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9077)

**Comment by MikeMcC399** [comment]
@Ein-Tim 

The test I did went like this:

**Day 1**
- I set the date to Day 2 and then immediately back to Day 1
- I got the error "Exposure check failed, with timestamp of Day 2"
- Later on Day 1 I got "Limit already reached"

**Day 2**
- It showed Updated: with Day 2 timestamp (which was really from the exposure check the day before)
- Several attempts were made to carry out an exposure check, but there was no success and no new timestamp

**Day 3**
- Exposure checks happened normally and the status Update: showed today's date and time

If a user has set the date to a long time in the future it looks like it will take a long time to recover after correcting the date. I would suggest to reset the app in this case as a workaround.

The app could be improved though. It should verify any saved `lastExposureDetectionTime` against a known correct server time and then discard it if it is in the future. That may not prevent the "Limit already reached" error occurring, but it should allow the app to recover faster from time travel and return from the future.
